### ISSUE TITLE
export to iOS not working

### ISSUE THREAD

**Issue by samtsejerome** [godotengine/godot#44577]
**Godot version: 3.2.3

xcode version:  11.4.1 (11E503a)

**running export on Macos catali
[gdscript2ios.zip](https://github.com/godotengine/godot/files/5726794/gdscript2ios.zip)
na:10.15.6

**Issue description:
after spending months on a prototype game intended for ios, I thought it was about time to try out the ios export from c# so I created a very simple project that does nothing but display a texture and move a sprite via a rigidbody2d and detect mouse clicks emulating tap to play a sound
but thafter spening an age creating the required images for apple, the export failed miserably (it hadnt created an .ipa after two hours of waiting (and the xcode project it generated looked nothing like the project running in godot )
I then asked a question in the godot q&a, link here:
https://godotengine.org/qa/90305/how-can-i-export-to-ios?show=90305#q90305
and got an answer that suggested 1.it should be possible/easy, and 2. my issue might be caused by c#script being in use(albeit a minute amount). so the next day (today) I created an even simpler project only using gdscript, and it also failed to create an ipa.   I only waited 20 minutes before giving up this time (as I didn't have more time available, (but it was stuck at the same 60% mark again-  )

Steps to reproduce:
1.create a simple godot project (see attached .zip file for the one I used)
2. select project>export
3. add iOS preset if not already present.
4.  enter valid team ID, bundle identifier and required icon filenames and an export path (outside of the project folder)
5. click export project and enter an archive filename then click save

**Minimal reproduction project:**

**Comment by naithar** [comment]
Exported your project on Godot 3.2.3 with my existing TeamID and BundleID of application that's on AppStore got this results:
<img width="239" alt="Снимок экрана 2020-12-26 в 20 47 41" src="https://user-images.githubusercontent.com/3750083/103156635-9a699800-47bb-11eb-8597-7fabd7902777.png">

So `ipa` and Xcode project seems to be created correctly.

Do you have an Xcode project at path that you export project to? If you do, you can open it with Xcode to check if there is a problem with signing identities or provisioning profiles that doesn't allow `ipa` to be built. 
Basically if you can build exported project with Xcode, Godot should be creating `ipa` and archive just fine on export. If `ipa` isn't created then there is some problem with settings that you've set while exporting. Checking with Xcode project can help you solve them.

**Comment by samtsejerome** [comment]
I tried again a couple more times  today with the same test project, and although  it created an xcode project, it didn't create an .ipa, nor did it provide any message to say export was finished (which is probably why I waited so long for it to finish the first time!)

I'm not an ios developer (but I'm using godot for that reason - and shouldn't need to be ?)
so opening the xcode project  wouldn't shed much light on the problem (as I wouldn't know what to look for)
also worth noting xcode hangs when I try to open the xcode project (it may be that the export process was still running when i tried to open it?, but thats a symptom of not have a clear 'export complete' message
<img width="407" alt="xcode-sshot" src="https://user-images.githubusercontent.com/68197715/103484279-77576d80-4de5-11eb-94ca-8d5b6fd4fa19.png">
<img width="493" alt="2ndattemptexport-sshot" src="https://user-images.githubusercontent.com/68197715/103484285-81796c00-4de5-11eb-848c-de684b589a6f.png">

<img width="619" alt="thirdattemptexport-sshot" src="https://user-images.githubusercontent.com/68197715/103484291-8ccc9780-4de5-11eb-996f-4e1e0f0b5166.png">


.)

**Comment by naithar** [comment]
>  opening the xcode project wouldn't shed much light on the problem (as I wouldn't know what to look for)

As I said you should just open the project and check what Xcode has issue with. Being or not being an iOS developer has noting to do with it as Xcode gives pretty clear errors. Even if error is not clear, it would still give some hints if it's a Godot issue (which I highly doubt). 

> I'm not an ios developer (but I'm using godot for that reason - and shouldn't need to be ?)

Godot is handing game development, so yeah you shouldn't be. But signing, verifying and uploading the resulting game depends on platform, especially on mobile. So having some experience with deployment platform might be crucial.

Judging from that I have managed to get `.ipa` file with my Team ID and Bundle ID, it seems like you have problems with Xcode and Apple Development certificates, not Godot itself. You probably haven't created provisioning profile, signing identity or bundle identifier, which should be created by youself at Apple Development websites. 

Xcode can actually handle this for you, so once again I recommend you to open the project and look what messages it gives. If the issue is with application signing, Xcode even gives a way to fix it with a single button press if iOS developer profile and other stuff has been setup for it.

As for Xcode not being responsive it might be the problem with iCloud synchronization that seems to be active for `gdscript2ios` folder. Or it could be your Mac not having enough power to handle Xcode and whatever app you are running with it at the same time.

**Comment by samtsejerome** [comment]
I tried again today with the same simple test project,  first double-checked everything in myapple account, but again got no .ipa in the output:
as shown here:
<img width="825" alt="3rdexportattempt-sshot" src="https://user-images.githubusercontent.com/68197715/104130009-65864500-5366-11eb-945d-d3b87c1deed7.png">
( I mistakenly called the folder 3rd ( because its actually the 4th attempt), apologies
then in desperation (because I dont want to have to learn xcode or ios dev to export) Iopened the project in xcode as suggested, and on trying to build, got a list of errors!? as per screenshot:
<img width="249" alt="xcode_builderrors-sshot" src="https://user-images.githubusercontent.com/68197715/104130090-bac25680-5366-11eb-9aed-8515ea5c4dc9.png">
I'm just using an official build from the godot site and a deliberately trivial project as a typical user might want to, and unable to produce an .ipa.

**Comment by Calinou** [comment]
@samtsejerome Are you using a Mono-enabled build? Even if you don't use any C# scripts, Mono export templates will still be used when exporting the project.

**Comment by samtsejerome** [comment]
I don’t know what that means tbh but I am aiming to ultimately export from c# ( was just trying gdscript first  )  I’ll try modifying ti c# script snd see if that works - thx 

Jerome Mayeux

**Comment by samtsejerome** [comment]
aplogies, I didnt realise mono was relevant as there is only gdscript in the project  ( I just had a quick look at the export presets available when i click add and there is just one  for iOs called iOs, so not really sure how that works, and its not covered in the docs, would  (  i may have installed an c# ios exporter at some stage (would I need to somehow remove that to be able to export gdscript ?)

**Comment by Calinou** [comment]
@samtsejerome To use non-Mono export templates, you need to export from a non-Mono editor.

**Comment by samtsejerome** [comment]
Good to know, ta.

Jerome Mayeux

**Comment by akien-mga** [comment]
Indeed, please try with with the standard build to see if the issue is specific to Mono. There used to be issues where exporting projects with no C# scripts with the Mono build would fail, but this was fixed AFAIK. But the iOS C# support is relatively new and might have other issues that are tripping you up.
